source/tasks/Metadata/task.json

```diff
@@ -0,0 +1,109 @@
+﻿{
+    "id": "b1861ef4-b62e-40c1-bcb0-be00d454a8a7",
+    "name": "OctopusMetadata",
+    "friendlyName": "Push Package Metadata to Octopus",
+    "description": "Collect metadata related to the build, including work items from commit messages, and push to your Octopus Deploy Server.",
+    "helpMarkDown": "set-by-pack.ps1",
+    "category": "Package",
+    "visibility": [
+        "Build",
+        "Release"
+    ],
+    "author": "Octopus Deploy",
+    "version": {
+        "Major": 4,
+        "Minor": 0,
+        "Patch": 0
+    },
+    "demands": [],
+    "minimumAgentVersion": "2.115.0",
+    "groups": [
+        {
+            "name": "advanced",
+            "displayName": "Advanced Options",
+            "isExpanded": false
+        }
+    ],
+    "inputs": [
+        {
+            "name": "OctoConnectedServiceName",
+            "type": "connectedService:OctopusEndpoint",
+            "label": "Octopus Deploy Server",
+            "defaultValue": "",
+            "required": true,
+            "helpMarkDown": "Octopus Deploy server connection"
+        },
+        {
+            "name": "Space",
+            "type": "pickList",
+            "label": "Space",
+            "defaultValue": "",
+            "required": true,
+            "properties": {
+                "EditableOptions": "True"
+            },
+            "helpMarkDown": "The space name within Octopus. You can leave this blank to use the default space. (Octopus Deploy 2019.1+)"
+        },
+        {
+            "name": "PackageId",
+            "type": "string",
+            "label": "Package ID",
+            "defaultValue": "",
+            "required": true,
+            "helpMarkDown": "The ID of the package. e.g. MyCompany.App"
+        },
+        {
+            "name": "PackageVersion",
+            "type": "string",
+            "label": "Package Version",
+            "defaultValue": "",
+            "required": true,
+            "helpMarkDown": "The version of the package; must be a valid [SemVer](http://semver.org/) version."
+        },
+        {
+            "name": "CommentParser",
+            "type": "pickList",
+            "label": "Work Items Source",
+            "defaultValue": "",
+            "required": false,
+            "helpMarkDown": "The source of work items or issues addressed by changes in the package",
+            "options": {
+                "-": "",
+                "Jira": "Jira",
+                "GitHub": "GitHub",
+                "Azure DevOps": "Azure DevOps"
+            }
+        },
+        {
+            "name": "Replace",
```

Comment:
we should update this to use the new overwrite mode feature Shannon implemented in the TeamCity/Bamboo plugins